bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/ConfigurationColumnViewItem.java

```diff
@@ -64,34 +67,66 @@ public String getTitle() {
     }
 
     public boolean hasChildren() {
+        if (isCommerceBucket())
+            return false;
+
         boolean isContainer = isConfigurationContainer();
-        return isContainer && hasCommerceSetting;
+        boolean hasChildren = resource.hasChildren();
+        boolean hasMoreChildren = StreamSupport.stream(resource.getChildren().spliterator(), false).count() > 1;
+        boolean hasSettings = resource.getChild("settings") != null;
+        return isContainer && (hasCommerceSetting || hasMoreChildren) || !isContainer && !hasCommerceSetting &&
+            (hasChildren && !hasSettings || hasMoreChildren && hasSettings);
+    }
+
+    private boolean isCommerceBucket() {
+        return resource.getPath().endsWith(Constants.COMMERCE_BUCKET_PATH);
     }
 
     public List<String> getQuickActionsRel() {
         List<String> actions = new ArrayList<>();
-
         if (isConfigurationContainer()) {
             // for /conf/<bucket>/settings folder we add the "Create" activator
             // so we can do a "client-side render condition"
-            actions.add(hasCommerceSetting || resource.getPath().equals(Constants.CONF_ROOT) ? "none"
-                : "cq-confadmin-actions-create-activator");
+            if (!hasCommerceSetting && !resource.getPath().equals(Constants.CONF_ROOT)) {
+                actions.add(CREATE_PULLDOWN_ACTIVATOR);
+                actions.add(CREATE_CONFIG_ACTIVATOR);
+            }
         }
 
-        if (!isConfigurationContainer()) {
+        if (isCommerceBucket()) {
             // for items which are not configuration containers (folders)
-            actions.add("cq-confadmin-actions-properties-activator");
-            actions.add("cq-confadmin-actions-delete-activator");
+            actions.add(PROPERTIES_ACTIVATOR);
+        } else {
+            if (!actions.contains(CREATE_PULLDOWN_ACTIVATOR)) {
+                actions.add(CREATE_PULLDOWN_ACTIVATOR);
+            }
+            actions.add(CREATE_FOLDER_ACTIVATOR);
+        }
+
+        if (isSafeToDelete()) {
+            actions.add(DELETE_ACTIVATOR);
         }
 
         return actions;
     }
 
     private boolean isConfigurationContainer() {
         return (resource.getPath()
-            .startsWith(Constants.CONF_ROOT) && (resource.isResourceType(JcrResourceConstants.NT_SLING_FOLDER) || resource.isResourceType(
-                JcrResourceConstants.NT_SLING_ORDERED_FOLDER))
+            .startsWith(Constants.CONF_ROOT) && (resource.isResourceType(JcrConstants.NT_FOLDER) ||
+                resource.isResourceType(JcrResourceConstants.NT_SLING_FOLDER) ||
+                resource.isResourceType(JcrResourceConstants.NT_SLING_ORDERED_FOLDER))
             && resource
                 .getChild(Constants.CONF_CONTAINER_BUCKET_NAME + "/" + Constants.CLOUDCONFIGS_BUCKET_NAME) != null);
     }
+
+    private boolean isSafeToDelete() {
+        return isCommerceBucket() || resource.getPath().startsWith(Constants.CONF_ROOT)
```

Comment:
Can we optimize this condition? Split similar conditions into boolean methods with meaningful names is one option.

The way it is now is barely readable.

Maybe I'm nitpicking, but I think `StreamSupport.stream(resource.getChild(Constants.CONF_CONTAINER_BUCKET_NAME).getChildren().spliterator(), false).count()` can be wrapped in a method to give it a more descriptive name.

Reply:
Yeah, those conditions have gotten a bit wild...